rundeckapp/grails-app/migrations/core/Execution.groovy

```diff
@@ -159,4 +162,61 @@ databaseChangeLog = {
             column(name: "workflow_id")
         }
     }
+
+    changeSet(author: "rundeckdev", id: "4.x-add-uuid-and-job-uuid") {
+        preConditions(onFail: "MARK_RAN") {
+            not {
+                columnExists(tableName: "execution", columnName: 'uuid')
+                columnExists(tableName: "execution", columnName: 'job_uuid')
+            }
+        }
+        addColumn(tableName: "execution") {
+            column(name: 'job_uuid', type: '${varchar255.type}')
+            column(name: 'uuid', type: '${varchar255.type}')
+        }
+
+    }
+
+    changeSet(author: "rundeckdev", id: "4.x-add-uuid-and-job-uuid-indexes") {
+        preConditions(onFail: "MARK_RAN"){
+            not {
+                indexExists(indexName: "execution_uuid_idx", tableName: "execution")
+                indexExists(indexName: "execution_job_uuid_idx", tableName: "execution")
+            }
+        }
+        createIndex(indexName: "execution_uuid_idx", tableName: "execution") {
+            column(name: "uuid")
+        }
+        createIndex(indexName: "execution_job_uuid_idx", tableName: "execution") {
+            column(name: "job_uuid")
+        }
+    }
+
+    changeSet(author: "rundeckdev", id: "4.x-populate-job-uuid") {
+        preConditions(onFail: "MARK_RAN") {
+            tableExists(tableName: "execution")
+            tableExists(tableName: "scheduled_execution")
+        }
+        sql("update execution set job_uuid = (select scheduled_execution.uuid from scheduled_execution where scheduled_execution.id = execution.scheduled_execution_id) where job_uuid is null")
```

Comment:
Maybe we need to document this in the release notes because this could take a couple of minutes with a large database when the APP will be updated

Reply:
I have removed the data population migration statements. That data is not required at this point, and if we require it in the future we can decide the best way to populate those columns at that point.